src/components/FirstRegistrationForm.tsx

```diff
@@ -0,0 +1,95 @@
+import { Formik, Form, Field } from 'formik';
+import Email from "./Email";
+import PasswordAndConfirm from './PasswordAndConfirm';
+import Checkbox from './Checkbox';
+import './FirstRegistrationForm.scss';
+import { isEmailExists } from '../services/api/isEmailExists';
+
+interface FormValues {
+  email: string;
+  password: string;
+  confirmPassword: string;
+  termsAndConditions: boolean;
+}
+
+interface FormErrors {
+  [key: string]: any
+}
+
+const emailInvalidationRules = [
+  /^\s*$/, // check string not empty
+  /^[^@]+$/, // @ should exist
+  /@[^@]*@/, // onle one @ is admissible
+  /^\./, // '.' can't be first symbol
+  /\.{2,}.+(?=@)/, // '.' can't repeat more than once in a row
+  /\.(?=@)/, // '.' can't be before @
+  /[^A-Za-z0-9_!#$%&'.*+\-/=?^`{|}~].*(?=@)/, // include only valid symbols before @
+  /(?<=@).*[^a-z0-9\-.]/, // include only valid symbols before @
+];
+
+const minSymbol = 6;
+const maxSymbol = 256;
+const passwordRegex = new RegExp("^[-/=!#$%&'*+?^_`{|}~.A-Z0-9]{" + minSymbol + "," + maxSymbol + "}$", "i");
+
+const FirstRegistrationForm = () => {
+  const initialValues: FormValues = {
+    email: '',
+    password: '',
+    confirmPassword: '',
+    termsAndConditions: false
```

Comment:
Please use 'is', 'has' prefixes for variables and constants of boolean type, e.g.

hasTermsAndConditions: false

This makes your code more readable

Reply:
fixed